### PR TITLE
Test cases - SCIMRoleManager, DefaultSCIMUserStoreErrorResolver, IdentityResourceURLBuilderTest

### PR DESCRIPTION
**Pull request by bhagyasakalanka**
Test cases for SCIMRoleManager, DefaultSCIMUserStoreErrorResolver, IdentityResourceURLBuilderTest

### CODE REVIEW

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java [hunk 1]
```diff
@@ -0,0 +1,1535 @@
...
+    public Object[][] dpCreateRoleExistingRole(){
+        return new Object[][] {
+                {"roleId1","roleDisplayName1","carbon.super","fail"},
+                {"roleId3","", "carbon.super","fail"}
+        };
+    }
+    @Test(dataProvider = "dpCreateRoleExistingRole")
+    public void testCreateRoleExistingRole(String roleId, String roleDisplayName,String tenantDomain, String expect)
+            throws IdentityRoleManagementException, BadRequestException, CharonException {
+        Role role = getDummyRole(roleId, roleDisplayName);
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java:86]: missing new line after the method signature

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,1535 @@
...
+@PrepareForTest({SCIMCommonUtils.class})
+public class SCIMRoleManagerTest extends PowerMockTestCase {
+    private final List<String> existingOrInvalidRoleIds = Arrays.asList("roleId1", "roleId3");
+    private final List<String> invalidRoleNames = Arrays.asList("system_roleName","system_testRoleName");
+    private final List<String> nonExistingRoleIds = Arrays.asList("roleId1", "roleId2");
+    private final List<String> invalidTenantDomainNames = Arrays.asList("invalidTenantDomain1", "invalidTenantDomain2");
+    private final List<String> existingRoleNames = Arrays.asList("newRoleName1", "newRoleName2");
+    private final List<String> nonExistingRoles = Arrays.asList("roleId1", "roleId2");
+    private final Set<String> systemRoles = new HashSet<>(Arrays.asList("roleDisplayName1", "roleDisplayName2"));
+
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java:60]: Shall we define two tenant domains as constants and reuse them .
```java
    private static final String SAMPLE_TENANT_DOMAIN = "carbon.super";
    private static final String SAMPLE_TENANT_DOMAIN2 = "abc.com";
```

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java [hunk 3]
```diff
@@ -0,0 +1,1716 @@
...
+        String result = "";
+        try {
+            Role createdRole = scimRoleManager.createRole(role);
+            assertEquals(createdRole.getDisplayName(), roleDisplayName);
+            assertEquals(createdRole.getId(), roleId);
+            result = SUCCESS;
+        }
+        catch (ConflictException e) {
+            if(e.getDetail().equals("Role with name: " + roleDisplayName +
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java:328]: Formatting issue

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java [hunk 4]
```diff
@@ -0,0 +1,1716 @@
...
+                    " already exists in the tenantDomain: "+ tenantDomain)) {
+                result = CONFLICT;
+            }else if(e.getDetail().equals("Role already exist for the role name: " + roleDisplayName)){
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java:331]: Formatting issue

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java (hunk 3 above)
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java:328]: Depending on the "detail" is not the best option.

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java [hunk 5]
```diff
@@ -0,0 +1,1716 @@
...
+                {SAMPLE_NON_EXISTING_ROLE_ID, SAMPLE_VALID_ROLE_NAME, "roleDomain1", SAMPLE_TENANT_DOMAIN, "attribute",
+                        true, FAIL},
+                {SAMPLE_NON_EXISTING_ROLE_ID2, SAMPLE_VALID_ROLE_NAME2, "roleDomainX", SAMPLE_INVALID_TENANT_DOMAIN,
+                        "attribute", false, FAIL}
+        };
+    }
+
+    @Test(dataProvider = "dpGetRoleNotFound")
+    public void testGetRoleNotFound(String roleId, String roleName, String domain, String tenantDomain,
+                                    String attribute, boolean attributeValue,
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java:357]: attributeValue is not actually the value, right?

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java [hunk 6]
```diff
@@ -0,0 +1,1716 @@
...
+            }
+            return null;
+        }).doNothing().when(mockRoleManagementService).deleteRole(roleId, tenantDomain);
+        SCIMRoleManager roleManager = new SCIMRoleManager(mockRoleManagementService, tenantDomain);
+        String result;
+        try {
+            roleManager.deleteRole(roleId);
+            verify(mockRoleManagementService, times(1)).deleteRole(roleId, tenantDomain);
+            result = SUCCESS;
+        }catch (NotFoundException e) {
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java:513]: Formatting issue

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java [hunk 7]
```diff
@@ -0,0 +1,1716 @@
...
+            return null;
+        }).doNothing().when(mockRoleManagementService).deleteRole(roleId, tenantDomain);
+
+        SCIMRoleManager roleManager = new SCIMRoleManager(mockRoleManagementService, tenantDomain);
+        String result;
+        try {
+            roleManager.deleteRole(roleId);
+            verify(mockRoleManagementService, times(1)).deleteRole(roleId, tenantDomain);
+            result = SUCCESS;
+        }catch (BadRequestException e) {
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java:547]: Formatting issue

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java [hunk 8]
```diff
@@ -0,0 +1,1716 @@
...
+            return null;
+        }).doNothing().when(mockRoleManagementService).deleteRole(roleId, tenantDomain);
+
+        SCIMRoleManager roleManager = new SCIMRoleManager(mockRoleManagementService, tenantDomain);
+        String result;
+        try {
+            roleManager.deleteRole(roleId);
+            verify(mockRoleManagementService, times(1)).deleteRole(roleId, tenantDomain);
+            result = SUCCESS;
+        }catch (CharonException e) {
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java:582]: Formatting issue

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java [hunk 9]
```diff
@@ -0,0 +1,1716 @@
...
+                thenReturn(roleBasicInfo);
+
+        SCIMRoleManager scimRoleManager = new SCIMRoleManager(mockRoleManagementService, tenantDomain);
+        String result;
+        try {
+            scimRoleManager.updateRole(oldAndNewRoles[0], oldAndNewRoles[1]);
+            result = SUCCESS;
+        } catch (BadRequestException e) {
+            result = BAD_REQUEST;
+        }catch (CharonException e) {
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java:1199]: Formatting issue

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java (hunk 5 above)
Comment (bhagyasakalanka) [reply components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java:357]: to call the getRole function, we need to provide a Map<String, Boolean>. Hence I create a dummy Map object using those attribute values.

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/DefaultSCIMUserStoreErrorResolverTest.java [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+
+    @Test(dataProvider = "dpResolveUserNotFound")
+    public void testResolveUserNotFound(String message, String checkMessage, int checkCode) {
+
+        UserStoreException userStoreException = new UserStoreException(message);
+        DefaultSCIMUserStoreErrorResolver defaultSCIMUserStoreErrorResolver = new DefaultSCIMUserStoreErrorResolver();
+        SCIMUserStoreException scimUserStoreException = defaultSCIMUserStoreErrorResolver.resolve(userStoreException);
+
+        assertEquals(checkCode, scimUserStoreException.getHttpStatusCode());
+        assertEquals(checkMessage, scimUserStoreException.getMessage());
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/DefaultSCIMUserStoreErrorResolverTest.java:56]: Let's remove errorMessage equal assertions. Messages can be changed from time to time.Remove the check in other places as well

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/DefaultSCIMUserStoreErrorResolverTest.java [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+    }
+
+    @DataProvider(name = "dpResolveUserNameMandatory")
+    public Object[][] dpResolveUserNameMandatory() {
+
+        return new Object[][]{
+                {"error", "32102", "Unable to create the user. Username is a mandatory field.", HttpStatus.SC_BAD_REQUEST},
+
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/DefaultSCIMUserStoreErrorResolverTest.java:64]: remove empty line

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/DefaultSCIMUserStoreErrorResolverTest.java [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+    @Test(dataProvider = "dpResolveInvalidDomainName")
+    public void testResolveInvalidDomainName(String message, String errorCode, String checkMessage, int checkCode) {
+
+        UserStoreException userStoreException = new org.wso2.carbon.user.core.UserStoreClientException(message, errorCode);
+        DefaultSCIMUserStoreErrorResolver defaultSCIMUserStoreErrorResolver = new DefaultSCIMUserStoreErrorResolver();
+        SCIMUserStoreException scimUserStoreException = defaultSCIMUserStoreErrorResolver.resolve(userStoreException);
+
+        assertEquals(checkCode, scimUserStoreException.getHttpStatusCode());
+        assertEquals(checkMessage, scimUserStoreException.getMessage());
+
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/DefaultSCIMUserStoreErrorResolverTest.java:97]: remove empty line

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/DefaultSCIMUserStoreErrorResolverTest.java [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+    @Test(dataProvider = "dpResolveUserNameMandatory")
+    public void testResolveUserNameMandatory(String message, String errorCode, String checkMessage, int checkCode) {
+
+        UserStoreException userStoreException = new org.wso2.carbon.user.core.UserStoreException(message, errorCode);
+        DefaultSCIMUserStoreErrorResolver defaultSCIMUserStoreErrorResolver = new DefaultSCIMUserStoreErrorResolver();
+        SCIMUserStoreException scimUserStoreException = defaultSCIMUserStoreErrorResolver.resolve(userStoreException);
+
+        assertEquals(checkCode, scimUserStoreException.getHttpStatusCode());
+        assertEquals(checkMessage, scimUserStoreException.getMessage());
+
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/DefaultSCIMUserStoreErrorResolverTest.java:77]: remove empty line

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/DefaultSCIMUserStoreErrorResolverTest.java [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+                break;
+            default:
+                userStoreException = new UserStoreException(message);
+                break;
+        }
+        DefaultSCIMUserStoreErrorResolver defaultSCIMUserStoreErrorResolver = new DefaultSCIMUserStoreErrorResolver();
+        SCIMUserStoreException scimUserStoreException = defaultSCIMUserStoreErrorResolver.resolve(userStoreException);
+
+        assertNull(scimUserStoreException);
+
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/DefaultSCIMUserStoreErrorResolverTest.java:129]: remove extra line

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/DefaultSCIMUserStoreErrorResolverTest.java [hunk 15]
```diff
@@ -0,0 +1,131 @@
...
+    }
+}
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/DefaultSCIMUserStoreErrorResolverTest.java:131]: add new line

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/IdentityResourceURLBuilderTest.java [hunk 16]
```diff
@@ -0,0 +1,147 @@
...
+    @Test
+    public void testBuildTenantQualifiedUrlsEnabledThrowURLBuilderException() throws NotFoundException {
+
+        mockStatic(IdentityTenantUtil.class);
+        when(IdentityTenantUtil.isTenantQualifiedUrlsEnabled()).thenReturn(true);
+        try {
+            when(mockServiceURLBuilder.build()).thenThrow(
+                    new URLBuilderException("Protocol of service URL is not available."));
+        } catch (URLBuilderException e) {
+            e.printStackTrace();
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/IdentityResourceURLBuilderTest.java:87]: add proper error log. don't use ` e.printStackTrace();`

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/IdentityResourceURLBuilderTest.java [hunk 17]
```diff
@@ -0,0 +1,147 @@
...
+    @Test
+    public void testBuildTenantQualifiedUrlsEnabled() throws NotFoundException, URLBuilderException {
+
+        mockStatic(IdentityTenantUtil.class);
+        when(IdentityTenantUtil.isTenantQualifiedUrlsEnabled()).thenReturn(true);
+        when(mockServiceURLBuilder.build()).thenReturn(mockServiceUrl);
+        when(mockServiceUrl.getAbsolutePublicURL()).thenReturn("www.public.url");
+        IdentityResourceURLBuilder identityResourceURLBuilder = new IdentityResourceURLBuilder();
+        String buildValue = identityResourceURLBuilder.build("resource");
+        System.out.println(buildValue);
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/IdentityResourceURLBuilderTest.java:74]: remove `System.out.println(buildValue);`

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/IdentityResourceURLBuilderTest.java [hunk 18]
```diff
@@ -0,0 +1,147 @@
...
+    @Test
+    public void testBuildTenantQualifiedUrlsEnabledThrowURLBuilderExceptionThrowNotFoundException() {
+
+        mockStatic(IdentityTenantUtil.class);
+        when(IdentityTenantUtil.isTenantQualifiedUrlsEnabled()).thenReturn(true);
+        try {
+            when(mockServiceURLBuilder.build()).thenThrow(
+                    new URLBuilderException("Protocol of service URL is not available."));
+        } catch (URLBuilderException e) {
+            e.printStackTrace();
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/IdentityResourceURLBuilderTest.java:106]: add a log and remove `e.printStackTrace();`

Comment (AnuradhaSK) [reply components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/IdentityResourceURLBuilderTest.java:106]: Check other places as well

**Review by madurangasiriwardena — APPROVED** [review]

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/DefaultSCIMUserStoreErrorResolverTest.java [hunk 19]
```diff
@@ -0,0 +1,77 @@
...
+import org.powermock.modules.testng.PowerMockTestCase;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.scim2.common.extenstion.SCIMUserStoreException;
+import org.wso2.carbon.user.api.UserStoreException;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNull;
+
+public class DefaultSCIMUserStoreErrorResolverTest extends PowerMockTestCase {
```
Comment (somindatommy) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/DefaultSCIMUserStoreErrorResolverTest.java:31]: Shall we add a class comment here even though this is a test class?

Comment (somindatommy) [reply components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/DefaultSCIMUserStoreErrorResolverTest.java:31]: Shall we fix in other places as well?

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java [hunk 20]
```diff
@@ -0,0 +1,1323 @@
...
+        };
+    }
+
+    @Test(dataProvider = "dataProviderForCreateRoleExistingRole")
+    public void testCreateRoleExistingRole(String roleId, String roleDisplayName)
+            throws IdentityRoleManagementException, BadRequestException, CharonException {
+
+        Role role = getDummyRole(roleId, roleDisplayName);
+        when(mockRoleManagementService.isExistingRole(anyString(), anyString()))
+                .thenReturn(true);
```
Comment (somindatommy) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java:145]: Can we move this to the above line?

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java [hunk 21]
```diff
@@ -0,0 +1,1323 @@
...
+    public void testGetRolePositive(String roleId, String roleName, String domain, String tenantDomain,
+                                    String attributeKey, Boolean attributeValue, boolean isEmptyLists)
+            throws IdentityRoleManagementException, BadRequestException, NotFoundException, CharonException {
+
+        org.wso2.carbon.identity.role.mgt.core.Role role = getDummyIdentityRole(roleId, roleName, domain, tenantDomain,
+                isEmptyLists);
+        Map<String, Boolean> attributeMap = null;
+        if (attributeKey != null) {
+            attributeMap = new HashMap<>();
+            attributeMap.put(attributeKey, attributeValue); //dummy details
```
Comment (somindatommy) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java:300]: Formatting error. Need to start from an upper case letter, there needs to be a space after `//` and need to end with a `.`. Better if we can add the comment in a new line.

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java [hunk 22]
```diff
@@ -0,0 +1,1323 @@
...
+                listRolesWithGET(null, startIndex, count, sortBy, sortOrder));
+    }
+
+    @Test
+    public void testListRolesWithGETCountNullZero()
+            throws NotImplementedException, BadRequestException, CharonException {
+
+        SCIMRoleManager roleManager = new SCIMRoleManager(mockRoleManagementService, SAMPLE_TENANT_DOMAIN);
+        List<Object> roles = roleManager.listRolesWithGET(null, 1, 0,
+                null, null);
```
Comment (somindatommy) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java:406]: Can we move this to the above line?

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java [hunk 23]
```diff
@@ -0,0 +1,1323 @@
...
+
+        Node rootNode = generateNodeBasedOnNodeType(nodeType, null);
+        when(mockRoleManagementService.getRoles(anyInt(), anyInt(), anyString(), anyString(), anyString())).
+                thenAnswer(invocationOnMock -> {
+                    Integer countArg = invocationOnMock.getArgumentAt(0, Integer.class);
+                    if (countArg != null && countArg < 0) {
+                        String errorMessage =
+                                "Invalid limit requested. Limit value should be greater than or equal to zero. limit: "
+                                        + count;
+                        throw new IdentityRoleManagementClientException(INVALID_LIMIT.getCode(), errorMessage);
```
Comment (somindatommy) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java:431]: Shall we format this in a nicer way?

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java [hunk 24]
```diff
@@ -0,0 +1,1323 @@
...
+                    return null;
+                });
+        when(mockRoleManagementService.getRoles(anyString(), anyInt(), anyInt(), anyString(),
+                anyString(), anyString())).
+                thenAnswer(invocationOnMock -> {
+                    Integer countArg = invocationOnMock.getArgumentAt(1, Integer.class);
+                    if (countArg != null && countArg < 0) {
+                        String errorMessage =
+                                "Invalid limit requested. Limit value should be greater than or equal to zero. limit: "
+                                        + count;
```
Comment (somindatommy) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java:442]: Shall we format this in a nicer way? 
```
String errorMessage ="Invalid limit requested. Limit value should be greater than or 
equal to zero. limit: "+ count;
```

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java [hunk 25]
```diff
@@ -0,0 +1,1323 @@
...
+        group3.setMember(u3);
+        Group group4 = new Group();
+        group4.setDisplayName("groupName4");
+        group4.setId("26d3a726-9c00-4f4c-8a4e-f5e310138084");
+        group4.setMember(u4);
+        Group group5 = new Group();
+        group5.setDisplayName("groupName5");
+        group5.setId("26d3a726-9c00-4f4c-8a4e-f5e310138085");
+        group5.setMember(u5);
+        //oldRole
```
Comment (somindatommy) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java:1093]: Formatting error.

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java [hunk 26]
```diff
@@ -0,0 +1,1323 @@
...
+        Role oldRole = new Role();
+        oldRole.setId(roleId);
+        oldRole.setDisplayName(oldRoleName);
+
+        //newRole
```
Comment (somindatommy) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java:1098]: Formatting error.

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java (hunk 21 above)
Comment (bhagyasakalanka) [reply components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java:300]: Fixed https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/387

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java (hunk 26 above)
Comment (bhagyasakalanka) [reply components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java:1098]: Fixed https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/387

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java (hunk 23 above)
Comment (bhagyasakalanka) [reply components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java:431]: Fixed https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/387

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/DefaultSCIMUserStoreErrorResolverTest.java (hunk 19 above)
Comment (bhagyasakalanka) [reply components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/DefaultSCIMUserStoreErrorResolverTest.java:31]: Fixed https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/387

**Pull request merged**